### DISCUSSION TITLE
Highlighted text

### DISCUSSION THREAD

**Discussion by Mano2731995** [pymupdf/PyMuPDF#2284] [Ideas]
Hai jorj

How to get and extract the highlighted text?

**Comment by JorjMcKie** [comment]
Assuming that the highlighting has happened via an annotation:
* loop over the annotations of the page (downselecting to appropriate annot type)
* take the annotation rectangle `annot.rect` and enlarge a little. This is recommended, because highlighting by many software packages create a too small annot rectangle to completely cover the complete text. So add an allowance like `rect = annot.rect + (-5, -5, 5, 5)` - which is 5 pixels large in every direction.
* Extract text like so `text = page.get_text(clip=rect)`.

If highlighting has **_not_** happened via annotations, things are much more complicated, so let's hope you will get away with this recipe.

**Reply by Mano2731995** [reply]
Hai jorj

I have tried below code for extracting the highlighted text 

Points= annot. rect 
a= page.get_text(clip=points) 

Expected result: need for only highlighted text

Actual result: Extracting highlighted text along with non highlighted text(pls refer image) 

![Input.jpg](https://user-images.githubusercontent.com/117537492/225680804-70e9ad0d-467a-4252-8e83-a858f6e35b90.jpg)

![Output.jpg](https://user-images.githubusercontent.com/117537492/225680872-f2e330e7-23a3-400e-9eed-1aa8b9326a74.jpg)

Pls suggest how to get only highlighted text

**Reply by JorjMcKie** [reply]
The annotation rectangle always is exactly this: a rectangle.
Your text overall however is two rectangles, each on its own line. You must use the `annot.vertices` property to determine those sub-areas.
